### PR TITLE
Feature/extern unquotes

### PR DESCRIPTION
**Pull request by b-studios**

### CODE REVIEW

**Comment by jiribenes** [comment]
Follow-up: Would be nice to have `${...}` nicely highlighted in VSCode.

**Comment by b-studios** [comment]
Language server seems to be broken.

File: effekt/shared/src/main/scala/effekt/Parser.scala [hunk 1]
```diff
@@ -147,16 +152,15 @@ class EffektParsers(positions: Positions) extends EffektLexers(positions) {
...
   lazy val externResource: P[Def] =
     (`extern` ~ `resource`) ~> (idDef ~ (`:` ~> blockType)) ^^ ExternResource.apply
 
-
-  lazy val externBody: P[String] =
-    ( multilineString
-    | s"(?!${multi})\"([^\"\n]*)\"".r // single-line strings
+  lazy val externBody: P[Template[Term]] =
+    ( multilineString ^^ { s => Template(List(s), Nil) }
+    | guard(regex(s"(?!${multi})".r)) ~> templateString(expr)
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/Parser.scala:163]: TODO we could make template strings also work in multiline strings

File: effekt/jvm/src/test/scala/effekt/ChezSchemeTests.scala [hunk 2]
```diff
@@ -26,6 +26,10 @@ abstract class ChezSchemeTests extends EffektTests {
     examplesDir / "pos" / "issue319.effekt",
     examplesDir / "pos" / "maps.effekt",
 
+    // splices not yet supported in Chez backends
+    examplesDir / "pos" / "capture" / "resources.effekt" ,
+    examplesDir / "pos" / "capture" / "ffi_blocks.effekt",
```
Comment (b-studios) [inline effekt/jvm/src/test/scala/effekt/ChezSchemeTests.scala:31]: We could make template strings also work in other backends.

File: effekt/shared/src/main/scala/effekt/core/Parser.scala [hunk 3]
```diff
@@ -71,7 +71,7 @@ class CoreParsers(positions: Positions, names: Names) extends EffektLexers(posit
     ( `extern` ~> externBody ^^ Extern.Include.apply
     | `extern` ~> (captures <~ `def`) ~ signature ~ (`=` ~> externBody) ^^ {
       case captures ~ (id, tparams, cparams, vparams, bparams, result) ~ body =>
-        Extern.Def(id, tparams, cparams, vparams, bparams, result, captures, body)
+        Extern.Def(id, tparams, cparams, vparams, bparams, result, captures, Template(List(body), Nil))
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/core/Parser.scala:74]: Right now template strings are not parsed in Core

File: effekt/shared/src/main/scala/effekt/core/PrettyPrinter.scala [hunk 4]
```diff
@@ -48,19 +48,26 @@ object PrettyPrinter extends ParenPrettyPrinter {
...
   def toDoc(e: Extern): Doc = e match {
     case Extern.Def(id, tps, cps, vps, bps, ret, capt, body) =>
-      "extern" <+> toDoc(capt) <+> "def" <+> toDoc(id) <+> "=" <+> paramsToDoc(tps, vps, bps) <> ":" <+> toDoc(ret) <+> "=" <+> "\"" <> body <> "\""
+      "extern" <+> toDoc(capt) <+> "def" <+> toDoc(id) <+> "=" <+> paramsToDoc(tps, vps, bps) <> ":" <+> toDoc(ret) <+> "=" <+> toDoc(body)
     case Extern.Include(contents) => emptyDoc // right now, do not print includes.
   }
 
+  def toDoc(t: Template[Pure]): Doc =
+    /// TODO
+    hsep(t.args.map(toDoc), comma)
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/core/PrettyPrinter.scala:69]: Right now template strings are not pretty printed nicely in core

File: effekt/shared/src/main/scala/effekt/lifted/PrettyPrinter.scala [hunk 5]
```diff
@@ -50,10 +50,14 @@ object PrettyPrinter extends ParenPrettyPrinter {
...
   def toDoc(e: Extern): Doc = e match {
     case Extern.Def(id, tparams, params, ret, body) =>
-      "extern def" <+> toDoc(id.name) <> signature(tparams, params, ret) <+> "=" <+> "\"" <> body <> "\""
+      "extern def" <+> toDoc(id.name) <> signature(tparams, params, ret) <+> "=" <+> "\"" <> toDoc(body) <> "\""
     case Extern.Include(contents) => emptyDoc // right now, do not print includes.
   }
 
+  // TODO implement
+  def toDoc(t: Template[Expr]): Doc =
+    hsep(t.args.map(toDoc), comma)
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/lifted/PrettyPrinter.scala:59]: Right now template strings are not pretty printed in lifted

**Pull request merged**